mono/utils/mono-merp.c

```diff
@@ -357,162 +331,112 @@ mono_init_merp (const intptr_t crashed_pid, const char *signal, MonoStackHash *h
+	FORMAT_TEXT_TO_FILE(handle, buff, sizeof (buff), "\t\t\"ExceptionType\" : \"%s\",\n", get_merp_exctype (merp->exceptionArg));
+	FORMAT_TEXT_TO_FILE(handle, buff, sizeof (buff), "\t\t\"StackChecksum\" : \"0x%llx\",\n", merp->hashes.offset_free_hash);
+	FORMAT_TEXT_TO_FILE(handle, buff, sizeof (buff), "\t\t\"StackHash\" : \"0x%llx\",\n", merp->hashes.offset_rich_hash);
 
 	// Provided by icall
-	mono_json_writer_indent (&writer);
-	mono_json_writer_object_key(&writer, "OSVersion:");
-	mono_json_writer_printf (&writer, "\"%s\",\n", merp->osVersion);
+	FORMAT_TEXT_TO_FILE(handle, buff, sizeof (buff), "\t\t\"OSVersion\" : \"%s\",\n", merp->osVersion);
+	FORMAT_TEXT_TO_FILE(handle, buff, sizeof (buff), "\t\t\"LanguageID\" : \"0x%x\",\n", merp->uiLidArg);
+	FORMAT_TEXT_TO_FILE(handle, buff, sizeof (buff), "\t\t\"SystemManufacturer\" : \"%s\",\n", merp->systemManufacturer);
+	FORMAT_TEXT_TO_FILE(handle, buff, sizeof (buff), "\t\t\"SystemModel\" : \"%s\",\n", merp->systemModel);
+	FORMAT_TEXT_TO_FILE(handle, buff, sizeof (buff), "\t\t\"EventType\" : \"%s\"\n", merp->eventType);
 
-	mono_json_writer_indent (&writer);
-	mono_json_writer_object_key(&writer, "LanguageID:");
-	mono_json_writer_printf (&writer, "\"0x%x\",\n", merp->uiLidArg);
-
-	mono_json_writer_indent (&writer);
-	mono_json_writer_object_key(&writer, "SystemManufacturer:");
-	mono_json_writer_printf (&writer, "\"%s\",\n", merp->systemManufacturer);
-
-	mono_json_writer_indent (&writer);
-	mono_json_writer_object_key(&writer, "SystemModel:");
-	mono_json_writer_printf (&writer, "\"%s\",\n", merp->systemModel);
-
-	mono_json_writer_indent (&writer);
-	mono_json_writer_object_key(&writer, "EventType:");
-	mono_json_writer_printf (&writer, "\"%s\"\n", merp->eventType);
-
-	// End of payload
-	mono_json_writer_indent (&writer);
-	mono_json_writer_object_end (&writer);
-	mono_json_writer_printf (&writer, "\n");
+	// End of parameters 
+	FORMAT_TEXT_TO_FILE(handle, buff, sizeof (buff), "\t}\n");
+	FORMAT_TEXT_TO_FILE(handle, buff, sizeof (buff), "}\n");
 
 	// End of object
-	mono_json_writer_indent_pop (&writer);
-	mono_json_writer_indent (&writer);
-	mono_json_writer_object_end (&writer);
-	
-	gchar *output = g_strdup (writer.text->str);
-	mono_json_writer_destroy (&writer);
-
-	return output;
+	close (handle);
+
+	return TRUE;
 }
 
-static gchar *
-mono_wer_template (MERPStruct *merp)
+static gboolean
+mono_write_wer_template (MERPStruct *merp)
 {
 	// Note about missing ProcessInformation block: we have no PID that makes sense
 	// and when mono is embedded and used to run functions without an entry point,
 	// there is no image that would make any semantic sense to send either. 
 	// It's a nuanced problem, each way we can run mono would need a separate fix.
 
-	GString *output = g_string_new ("");
+	int handle = g_open (merp->werXmlPath, O_WRONLY | O_CREAT | O_TRUNC, merp_file_permissions);
+	if (handle == -1) {
```

Comment:
`g_assertf (handle != -1, "Could not open WER XML file at %s", merp->merpFilePath);`